File: src/main/java/com/example/Ok.java
package com.example;

/** Successful result value. */
public final class Ok<T> implements Result<T> {
    private final T value;

    Ok(T value) {
        this.value = value;
    }

    @Override
    public boolean isOk() {
        return true;
    }

    @Override
    public Option<T> value() {
        return new Some<>(value);
    }

    @Override
    public Option<String> error() {
        return new None<>();
    }
}

File: src/main/node/com/example/Ok.ts
/** Successful result value. */
export default class Ok<T> implements Result<T> {
    private readonly value: any;

    Ok(value: any): void {
        // TODO
    }

    @Override
    isOk(): boolean {
        return /* TODO */;
    }

    @Override
    value(): Option<any> {
        return /* TODO */;
    }

    @Override
    error(): Option<string> {
        return /* TODO */;
    }
}
